src/Features/CSharp/Portable/BraceMatching/BlockCommentBraceMatcher.cs

```diff
@@ -14,33 +14,52 @@
 namespace Microsoft.CodeAnalysis.CSharp.BraceMatching;
 
 [ExportBraceMatcher(LanguageNames.CSharp), Shared]
-internal class BlockCommentBraceMatcher : IBraceMatcher
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class BlockCommentBraceMatcher() : IBraceMatcher
 {
-    [ImportingConstructor]
-    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-    public BlockCommentBraceMatcher()
-    {
-    }
-
     public async Task<BraceMatchingResult?> FindBracesAsync(Document document, int position, BraceMatchingOptions options, CancellationToken cancellationToken)
     {
         var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+        var token = root.FindToken(position, findInsideTrivia: false);
+        if (token == default)
+            return null;
+
+        var span = token.Span;
+        if (span.Contains(position))
+            return null;
+
         var text = await document.GetValueTextAsync(cancellationToken).ConfigureAwait(false);
-        var trivia = root.FindTrivia(position);
```

Comment:
it seems like findTrivia is just not that very good.  Looking at the impl, it's pretty darn complex.  FindToken seems to be much simpler and much more optimized.  So changing our code here to just get the token, and walk the start/end trivia of it

Reply:
I've never really internalized how structured trivia works.

Nodes can have child nodes/tokens
Tokens have leading/trailing trivia
Trivia may have structure (which is a node)

Can those nodes inside the trivia repeat this cycle?

Reply:
> Nodes can have child nodes/tokens
> Tokens have leading/trailing trivia
> Trivia may have structure (which is a node)

100% correct.

>Can those nodes inside the trivia repeat this cycle?

No.  trivia only goes one more level deep.  So within a top levle trivia (like a `#if false` directive trivia), you have:

1. Nodes (the directive trivia, its expressions etc)
2. tokens 9the `#`, `if` and `false` tokens
3. trivia (the whitespace between those)

However, those final trivia are *never* structured themselves,  so it bottoms out.

We were explicit int he syntax design that no operations go from top level to strucutred trivia automatically, and you always pass the `findInsideTrivia` flag to control that.  And, because there's no more recursion, there's no confusion abuot which level you're diving down into. It's either the top level, or the single structured-trivia level.

Does that make sense?